### PR TITLE
python3Packages.pygraphviz: 1.6 -> 1.7

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pygraphviz/default.nix [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
-
   patches = [
-    # pygraphviz depends on graphviz being in PATH. This patch always prepends
-    # graphviz to PATH.
+    # pygraphviz depends on graphviz executables and wc being in PATH
     (substituteAll {
-      src = ./graphviz-path.patch;
-      inherit graphviz;
+      src = ./path.patch;
+      path = lib.makeBinPath [ graphviz coreutils ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/pygraphviz/default.nix:28]: Note that wc actually doesn't seem to be used: https://github.com/pygraphviz/pygraphviz/issues/332

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113040` at adb7d7ff run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/c7b74ee7f1eced3996b3627ec61d4ca2">python37Packages.cirq</a></li></li>
    <li><a href="https://gist.github.com/0029a6d75ba673e92349838a1cd9fbaa">python37Packages.pygraphviz</a></li></li>
    <li><a href="https://gist.github.com/08ef1536b1129b8ceec0c7fa030c8014">python38Packages.cirq</a></li></li>
    <li><a href="https://gist.github.com/872f2209cb02ba5b7d09c8f503beecbd">python38Packages.pygraphviz</a></li></li>
    <li><a href="https://gist.github.com/d540bfa7595b31ed57b8fa439c40dbaf">python39Packages.cirq</a></li></li>
    <li><a href="https://gist.github.com/5d3d6e8780766c0ae5b83e6c93015d7e">python39Packages.pygraphviz</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113040` at adb7d7ff run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/262799b2cffdc9dfca2dd4366bfc3acb">python37Packages.cirq</a></li></li>
    <li><a href="https://gist.github.com/9c1ad70b9fb5b96fbf4f7a850f79033e">python37Packages.pygraphviz</a></li></li>
    <li><a href="https://gist.github.com/fd2f5f1e8136f74bcb0cf5d5104365a2">python38Packages.cirq</a></li></li>
    <li><a href="https://gist.github.com/5f89705771570039239585b2106050e0">python38Packages.pygraphviz</a></li></li>
    <li><a href="https://gist.github.com/b64eab11ff50831dfd5aca2e0b8d194c">python39Packages.pygraphviz</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/pygraphviz/default.nix [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
+  buildInputs = [
+    graphviz
+  ];
+
+  checkInputs = [
+    pytest
+  ];
+
+  checkPhase = ''
+    pytest --pyargs pygraphviz
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pygraphviz/default.nix:45]: Can this be done with pytestCheckHook?

File: pkgs/development/python-modules/pygraphviz/default.nix [hunk 3]
```diff
@@ -1,40 +1,54 @@
...
+    pkg-config
+  ];
+
+  buildInputs = [
+    graphviz
+  ];
+
+  checkInputs = [
+    pytest
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pygraphviz/default.nix:42]:
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ graphviz ];

  checkInputs = [ pytest ];
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113040` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>python37Packages.cirq</li>
<li>python37Packages.pygraphviz</li>
<li>python38Packages.cirq</li>
<li>python38Packages.pygraphviz</li>
<li>python39Packages.pygraphviz</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.cirq:
<details>


Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/cirq/default.nix:6:3:

```
  |
6 | , fetchpatch
  |   ^
```
</details>
python38Packages.cirq:
<details>


Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/cirq/default.nix:6:3:

```
  |
6 | , fetchpatch
  |   ^
```
</details>

File: pkgs/development/python-modules/pygraphviz/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pygraphviz/default.nix:45]: I tried `pytestFlagsArray`, but it didn't seem to work.

**Pull request merged**